Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1342,39 +1339,296 @@ namespace OpenInfraPlatform
+				* \param[in] senseAgreement			Does the resulting geometry have the same sense agreement as the \c IfcOffsetCurveByDistances.
+				*
+				* \note The function is not implemented.
+				* \internal TODO.
+				*/
+				void convertIfcOffsetCurveByDistances(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcOffsetCurveByDistances>& offsetCurveByDistances,
+					std::vector<carve::geom::vector<3>>& targetVec,
+					std::vector<carve::geom::vector<3>>& segmentStartPoints,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim1Vec,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim2Vec,
+					const bool senseAgreement,
+					const typename IfcEntityTypesT::IfcTrimmingPreference& trimmingPreference
+				) const noexcept(false)
+				{
+					// **************************************************************************************************************************
+					//	https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/lexical/IfcOffsetCurveByDistances.htm
+					//	ENTITY IfcOffsetCurveByDistances
+					//		SUBTYPE OF(IfcOffsetCurve);
+					//			OffsetValues: LIST[1:? ] OF IfcPointByDistanceExpression;
+					//			Tag: OPTIONAL IfcLabel;
+					//	END_ENTITY;
+					// **************************************************************************************************************************
+
+					EXPRESSReference<typename IfcEntityTypesT::IfcCurve> basisCurve = offsetCurveByDistances->BasisCurve;
+					std::vector<carve::geom::vector<3>> basisCurvePoints;
+
+					convertIfcCurve(basisCurve, basisCurvePoints, segmentStartPoints,
+						trim1Vec, trim2Vec, senseAgreement, trimmingPreference);
+
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcPointByDistanceExpression>> OffsetValues = offsetCurveByDistances->OffsetValues;
+					std::vector < std::vector<carve::geom::vector<3>>> offsetCurves;
+
+					for (auto OffsetValue : OffsetValues) {
+						std::vector<carve::geom::vector<3>> offsetCurvePoints;
+						double distance = 0.;
+						switch (OffsetValue->DistanceAlong.which())
+						{
+						case 0:
+						{
+							// Calculate the length using \c IfcNonNegativeLengthMeasure. 
+							distance = OffsetValue->DistanceAlong.get<0>();// *this->UnitConvert()->getLengthInMeterFactor();
+							break;
+						}
+						case 1:
+						{
+							// Calculate the length using \c IfcParameterValue.
+							distance = OffsetValue->DistanceAlong.get<1>();
+							break;
+						}
+						default:
+							throw oip::InconsistentGeometryException(OffsetValue, "Could not determine DistanceAlong!");
+						}
+						
+						double offsetLateral = OffsetValue->OffsetLateral * this->UnitConvert()->getLengthInMeterFactor();
+						double offsetVertical = OffsetValue->OffsetVertical * this->UnitConvert()->getLengthInMeterFactor();
+						double offsetLongitudinal = OffsetValue->OffsetLongitudinal * this->UnitConvert()->getLengthInMeterFactor();

```

Comment:
use them as discussed yesterday
It's important to apply them in the order that is specified in the IFC documentation